pkg/auth/providers/oidc/oidc_provider.go

```diff
@@ -46,9 +46,9 @@ type OpenIDCProvider struct {
 type ClaimInfo struct {
 	Subject           string   `json:"sub"`
 	Name              string   `json:"name"`
-	PreferredUsername string   `json:preferred_username`
-	GivenName         string   `json:given_name`
-	FamilyName        string   `json:family_name`
+	PreferredUsername string   `json:"preferred_username"`
+	GivenName         string   `json:"given_name"`
+	FamilyName        string   `json:"family_name"`
```

Comment:
Not sure how this struct is used but this will change the JSON representation.
Since before this change, when the tags were invalid, the default key was `preferredUsername`, and now it will be `perferred_username`
```suggestion
	PreferredUsername string   `json:"preferred_username"`
	GivenName         string   `json:"given_name"`
	FamilyName        string   `json:"family_name"`
```

Reply:
Not quite sure what the suggestion was here.
But yes, naturally the representation will be changed, which we want as it's expected to be in snake case AFAIK. And we want to fix the issue with the tags themselves.

Reply:
Also as far as I can tell these particular three fields are not being used anywhere (yet?).

Reply:
> the default key was `preferredUsername`

The default key was `PreferredUsername`, by default without json tags.

But other than that, I couldn't find any uses for those either, which is why I suggested we remove, but I'm not against keeping them if we know we'll be using them in the future :shrug:

Reply:
These correspond to a subset of standard claims, but don't appear to include all of the fields: https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims

When requested as part of UserInfo, we expect to always see the `sub`ject identifier, but all other claims are optional even if requested via scopes: https://openid.net/specs/openid-connect-core-1_0.html#UserInfoResponse

Claims will arrive as `snake_case` if they are present at all, so `CamelCase` is certainly incorrect. Since these fields were almost certainly not being unmarshalled correctly in the first place, and no Rancher code seems to be using them, this is a strong argument in favor of their removal. It doesn't _look_ like the ClaimsInfo struct is meant to hold all possible values that we might see from the provider, just the ones Rancher cares about.